api/document/json_importer/importer.py

```diff
@@ -0,0 +1,131 @@
+from collections import defaultdict
+from typing import (Callable, DefaultDict, Dict, List, NamedTuple, Optional,
+                    Type)
+
+from django.db import transaction
+
+from document.models import Annotation, DocNode, ExternalLink, FootnoteCitation
+from document.tree import JSONAwareCursor, JsonDict
+from reqs.models import Policy
+
+
+class PreparedNode(NamedTuple):
+    node: JsonDict
+    content: List[JsonDict]
+    children: List[JsonDict]
+
+    @classmethod
+    def from_json(cls, node):
+        prepped = {}
+        for key in PRIMITIVE_DOC_NODE_FIELDS:
+            if key in node:
+                prepped[key] = node[key]
+        content = node.get('content', [])
+        prepped['text'] = get_content_text(content)
+        return cls(
+            node=prepped,
+            content=content,
+            children=node.get('children', []),
+        )
+
+
+PRIMITIVE_DOC_NODE_FIELDS = [
+    # TODO: Include 'identifier' here?
+    'node_type',
+    'type_emblem',
+    'marker',
+    'title',
+]
+
+Annotator = Callable[
+    [JSONAwareCursor, JsonDict, int],
+    Annotation
+]
+
+annotators: Dict[str, Annotator] = {}
+
+
+def annotator(fn: Annotator):
+    annotators[fn.__name__] = fn
+    return fn
+
+
+@annotator
+def footnote_citation(cursor: JSONAwareCursor, content: JsonDict,
+                      start: int) -> FootnoteCitation:
+    text = content['text']
+    referencing = list(cursor.filter(
+        lambda m: m.node_type == 'footnote'
+        and m.type_emblem == text.strip()
+    ))
```

Comment:
This might be copy-pasted from somewhere else, but if we have the referenced node's identifier, we should be able to `cursor.jump_to` it directly.

Reply:
Yep, it's copy-pasted from the XML importer annotation code.

Hmm, I'm not sure if we do actually have the referenced node's identifier, since the identifiers change based on where they happen to be in the document structure? Like, won't the identifier be something like `policy_1__sec_1__para_1__footnote_1`?

Reply:
That's a good point. Let's ponder it a bit more -- I don't think it's safe to assume a footnote marker will be unique, but we aren't generating the identifiers in the editor, so there's no way to refer to another node. This's likely to come up if we add in-document definitions or internal links, but we can punt for now.